osbrain/agent.py

```diff
@@ -754,15 +781,51 @@ def _process_sub_event(self, socket, serialized):
     def str2bytes(self, message):
         return message.encode('ascii')
 
+    def _is_address_internal(self, address):
+        """
+        TODO Perhaps we want to manually create inprocess addresses.
+            - Keep a variable for each one specifying whether it was internal,
+              regardless of type?
+
+        Returns
+        -------
+        bool
+            Whether the address is used for internal communication.
+        """
+        return address in ('loopback', '_loopback_safe', 'inproc://loopback',
+                           'inproc://_loopback_safe')
+
     def send(self, address, message, topic=''):
+        """
+        TODO
+        """
         assert isinstance(topic, str), 'Topic must be of `str` type!'
-        serialized = pickle.dumps(message, -1)
+        # Check if socket is for internal use
+        if self._is_address_internal(self.address[address]):
```

Comment:
Here again, we should get the information on the serializer type from the `AgentAddress`, not by looking if its "internal" or not (internal sockets should have serialization set as `raw`, so it will work for them as well.